build.ps1

```diff
@@ -36,4 +36,4 @@ $SDKPATH = "$CLIPATH\sdk\$SdkVersion"
 
 $captured_args = $args
 
-Exec-Block { & "$CLIPATH\dotnet" "$SDKPATH/MSBuild.dll" "$SCRIPT_ROOT/build.proj" /flp:v=detailed /clp:v=detailed $captured_args } | Out-Host
+Exec-Block { & "$CLIPATH\dotnet" "$SDKPATH/MSBuild.dll" "$SCRIPT_ROOT/build.proj" /flp:v=diag /clp:v=m $captured_args } | Out-Host
```

Comment:
`diag` is going to get really, really big, isn't it?

Reply:
Maybe instead we could pass in `/bl` for binlog?  http://msbuildlog.com/

Reply:
it will only be diag for the source-build itself not the other repos.

Reply:
I don't want to force us to use a custom tool to read the logs, especially on non-windows.

Reply:
Using minimum for the console output doesn't work well in CI, we currently don't capture the log files at all.

If we are going to change the console output, can you also ensure we can get access to the log files in CI?

I think this is how you do it:  https://github.com/dotnet/core-setup/blob/2e5ec4b9ac8084c61fc4b8130c33f08f783aaae3/netci.groovy#L104-L108

Reply:
> I don't want to force us to use a custom tool to read the logs, especially on non-windows.

Can we do both?  I don't know how much you've tried msbuildlog, but it is rather fantastic.

Reply:
That is related to my other question. I would prefer for CI we just dump things to the console so there is no need for finding logs and downloading them. So perhaps we should have pass the extra detailed switch in CI to make it have higher logging.

Reply:
>Can we do both? I don't know how much you've tried msbuildlog, but it is rather fantastic.

We could do both on windows I suppose. I tried it a number of times for corefx but the logs are so huge it cannot handle reading those diag logs. I personally find I always need diag logs to get property values when diagnosing most issues.

Reply:
It also looks like msbuild supports "replaying" the binary log to a diag log file:  https://github.com/Microsoft/MSBuild/wiki/Binary-Log#replaying-a-binary-log.  Which means we could do binlog, and then if someone needed to view to the diag file log, they could just replay it into a text file.

Reply:
> I would prefer for CI we just dump things to the console so there is no need for finding logs and downloading them.

For large logs, my browser can't display and search very well.  So I normally just download the console output anyway and then use a text editor to read/search it.

Reply:
Looking at the logs from your run, it doesn't appear to get getting any logs from the sub-repos:

```
14:48:18   [21:48:18.56] Building 'common'
14:48:18   Running command:
14:48:18      
14:48:18     Log: /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/bin/logs/common.log
14:48:18     With Enivornment Varibles:
14:48:18       DOTNET_RUNTIME_ID=rhel.7-x64
14:48:18       DOTNET_TOOL_DIR=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/dotnetcli/
14:48:18       BUILD_TOOLS_TOOL_DIR=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/
14:48:18       BUILDTOOLS_SKIP_CROSSGEN=1
14:48:18       DotNetBuildFromSource=true
14:48:18       DotNetRestorePackagesPath=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/packages/
14:48:18       DotNetCoreSdkDir=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/dotnetcli/
14:48:18       DotNetBuildToolsDir=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/
14:48:24   [21:48:24.26] Building 'common'...done
14:48:24   [21:48:24.27] Building 'nuget-client'
14:48:24   Running command:
14:48:24      
14:48:24     Log: /mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/bin/logs/nuget-client.log
14:48:24     With Enivornment Varibles:
14:48:24       DOTNET_RUNTIME_ID=rhel.7-x64
14:48:24       DOTNET_TOOL_DIR=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/dotnetcli/
14:48:24       BUILD_TOOLS_TOOL_DIR=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/
14:48:24       BUILDTOOLS_SKIP_CROSSGEN=1
14:48:24       DotNetBuildFromSource=true
14:48:24       DotNetRestorePackagesPath=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/packages/
14:48:24       DotNetCoreSdkDir=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/dotnetcli/
14:48:24       DotNetBuildToolsDir=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/
14:48:24       MS_PFX_PATH=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/keys/NuGet.Client.snk
14:49:15   [21:49:15.15] Building 'nuget-client'...done
```

You should try to make one fail to see what the debugging experience looks like.

Reply:
It also really stinks when you look at the console output to see how far along the build is, and it just sits there for 9 minutes doing nothing:

```
14:49:15       DotNetBuildToolsDir=/mnt/resource/j/workspace/dotnet_source-build/dev_release_2.0/RHEL7.2_Release_prtest/Tools/
14:58:20   [21:58:20.20] Building 'coreclr'...done
```

Reply:
As we chatted I fixed this in CI to have more logging but for local dev scenarios I like the cleaner logs but folks can enable more logging if they like.